common/djangoapps/student/views.py

```diff
@@ -894,6 +894,23 @@ def dashboard(request):
     return response
 
 
+def get_refund_status(request):
+    # get course key
+    course_key = request.GET.get('course_id')
+    if not course_key:
+        course_enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+
+        user_already_has_certs_for = GeneratedCertificate.course_ids_with_certs_for_user(request.user)
+        refundable_status = True if course_enrollment.refundable(
+                user_already_has_certs_for=user_already_has_certs_for
+            ) else False
+
+        from rest_framework.response import JsonResponse
+        # use json to pass the result to ajax call
+        return JsonResponse({'course_refundable_status': refundable_status})
```

Comment:
Instead of returning a json response, why can't we return a view for the dialog instead?
We can put the dialog content within an iframe in the pop up model, right?